src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -257,6 +263,13 @@ protected function getGroups(array $context): array
         return is_scalar($groups) ? (array) $groups : $groups;
     }
 
+    protected function getAttributeGroups(AttributeMetadataInterface $attributeMetadata): array
```

Comment:
This is on the hot patch. Introducing a new function will have a significant performance impact when serializing a large set of data. Also, we try to not add new protected methods in this class (we're in the process of splitting it and using composition instead of inheritance).

Could you inline it? By the way, you should also inline the one in the extractor.